gui/Network/Network.cpp

```diff
@@ -0,0 +1,105 @@
+#include "Network.hpp"
+#include "logging/Logger.hpp"
+
+#include <arpa/inet.h>
+#include <array>
+#include <netinet/in.h>
+#include <poll.h>
+#include <stdexcept>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <vector>
+
+Network::Network(int port, const std::string &hostname)
+  : _port(port), _hostname(hostname)
+{
+  _fdServer = socket(AF_INET, SOCK_STREAM, 0);
+  if (_fdServer < 0) {
+    throw std::
+      runtime_error("Error: socket, Function: Network, File: Network.cpp");
+  }
+}
+
+void Network::RunNetwork()
+{
+  sockaddr_in server_addr{};
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_port = htons(_port);
+
```

Comment:
To use correcly connect
```suggestion
sockadrr_len server_addr_len;
```

Reply:
done